app/k8s-api-client.js

```diff
@@ -0,0 +1,36 @@
+const config = require('./config');
+const { APIClient, JWTAuth } = require('./api-client');
+const url = require('url');
+
+
+class KubernetesAPIClient extends APIClient {
+  endpoint_url(endpoint, namespace = undefined) {
+    if (!endpoint) {
+      throw new Error('Missing endpoint');
+    }
+
+    let ns = namespace;
```

Comment:
Couldn't this logic be replaced with something like:

```javascript
let ns = this.namespace || 'default';
```

Another couple of things to bear in mind:
1) We may want to perform certain operation (as a superuser) in other users' namespaces, e.g. Robin wants to restart a user rstudio instance.
2) We may use this client to make requests for non-namespaces things, e.g. get the list of namespaces, although maybe not yet (just a thought)

Reply:
No, it's more like `let ns = namespace || this.namespace || 'default';` - maybe I should change it to that.

Reply:
The `namespace` parameter can be overridden for superusers accessing other users' namespaces, although probably the easiest way would be to set `api.namespace` before making the request.

Reply:
This code might need to be revisited if we have to list namespaces, but YAGNI

Reply:
Agreed.